zeroex/orderwatch/order_watcher.go

```diff
@@ -1178,6 +1178,7 @@ func (w *Watcher) convertValidationResultsIntoOrderEvents(
 			timestampIsValid := order.ExpirationTimeSeconds.Cmp(validationBlockTimestampSeconds) == 1
 			isOrderUnexpired := order.IsRemoved && timestampIsValid
 
+			// If order was previously expired, check if it has become unexpired.
```

Comment:
I think this comment belongs a line or two above. At this point we are not checking if the order has become unexpired, we _have_ already checked if the order has become unexpired (it has) and now we are emitting the appropriate events and updating the database based on that.

I also think we should add a little bit more context to the comment. In particular, it's important to note the relationship between fillable amount and whether an order was expired. The check for the fillable amount occurs above, so without a comment, the connection may not be clear. Something like:

```suggestion
			// We can tell that an order was previously expired if it was marked as removed with a non-zero fillable amount. There is no other explanation for this database state. The order is considered "unexpired" if it was previously expired but now has a valid expiration time based on the latest block timestamp.
```